### PR TITLE
feat: handle nested and nullable lists

### PR DESCRIPTION
**Pull request by Noroth**
## Summary by CodeRabbit

* **New Features**
  * Added extensive support for nullable and nested list fields across Projects, Tasks, Milestones, Employees, and Products.
  * Introduced new GraphQL queries for project tags, archived projects, tasks grouped by priority, and resource matrices.
  * Enhanced schema with new fields such as tags, dependencies, alternative projects, resource groups, skills, certifications, project history, and feature matrices.

* **Bug Fixes**
  * Improved handling of nil and empty list scenarios in queries and test assertions.

* **Refactor**
  * Updated data models and helper methods to support complex list structures and richer metadata.
  * Renamed certain functions for consistent naming.

* **Tests**
  * Expanded test coverage for nested lists, nullable fields, and edge cases in queries and responses.
  * Added new tests for tasks by priority, resource matrices, and employee project history.

* **Documentation**
  * Clarified GraphQL-to-Protobuf conversion rules for nullable and nested lists.

* **Chores**
  * Updated dependencies to newer versions.

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Noroth** [comment]
@coderabbitai summary

File: demo/pkg/subgraphs/projects/src/main_test.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func setupTestService(t *testing.T) *testService {
 	// Start the server
 	go func() {
 		if err := grpcServer.Serve(lis); err != nil {
-			t.Fatalf("failed to serve: %v", err)
+			t.Logf("failed to serve: %v", err)
 		}
```
[the remark this answers is not in the document]
Comment (Noroth) [reply demo/pkg/subgraphs/projects/src/main_test.go]: You cannot call t.Fatal, FailNow etc in a sub goroutine within a test.

**Review by devsergiy — APPROVED** [review]
